grpc-server/src/main/java/com/netflix/conductor/grpc/server/GRPCServerProvider.java

```diff
@@ -43,15 +46,25 @@ public GRPCServerProvider(
     @Override
     public Optional<GRPCServer> get() {
         return configuration.isEnabled() ?
-                Optional.of(
-                        new GRPCServer(
-                                configuration.getPort(),
-                                healthServiceImpl,
-                                eventServiceImpl,
-                                metadataServiceImpl,
-                                taskServiceImpl,
-                                workflowServiceImpl
-                        ))
+                Optional.of(getGRPCServer())
                 : Optional.empty();
     }
+
+    private GRPCServer getGRPCServer() {
+        ImmutableList.Builder<BindableService> services = ImmutableList.<BindableService>builder().add(
+                healthServiceImpl,
+                eventServiceImpl,
+                metadataServiceImpl,
+                taskServiceImpl,
+                workflowServiceImpl);
+
+        if (configuration.isReflectionEnabled()) {
+            services.add(ProtoReflectionService.newInstance());
+        }
+
+        return new GRPCServer(
+                configuration.getPort(),
+                services.build()
```

Comment:
Related to the above question about varargs, I am pretty sure you could have just passed an Array here rather than change the signature. /shrug